### DISCUSSION TITLE
How do I start Keystone to listen on a different port number?

### DISCUSSION THREAD

**Discussion by shivathejas** [keystonejs/keystone#7696] [Questions] [answered]
How to run the Keystone on a different port?

**Comment by pahaz** [comment]
You can use `PORT` https://github.com/keystonejs/keystone/blob/78193f9c9d93655fb0d4b8dc494fbe4c622a4d64/packages/keystone/constants.js#L2 

And change `package.json` file scripts like so:

![image](https://user-images.githubusercontent.com/1394025/79560512-65e00100-80b0-11ea-9a41-f2e8226f953b.png)

Or you can use `--port` argument https://github.com/keystonejs/keystone/blob/78193f9c9d93655fb0d4b8dc494fbe4c622a4d64/packages/keystone/bin/utils.js#L60

![image](https://user-images.githubusercontent.com/1394025/79560763-ddae2b80-80b0-11ea-8e56-a63d973767ae.png)

**Comment by kilinkis** [comment]
is this still true for v6 @pahaz? none of the above methods worked for me

**Comment by machineghost** [comment]
Same here:

    "dev": "keystone dev --port=3001",

still runs on port 3000.

**Comment by machineghost** [comment, marked as answer]
... but what I just discovered _does_ work is setting a `PORT` environmental variable:

    "dev": "PORT=3001 keystone dev",

Now when I `npm run dev` I see:

>⭐️ Dev Server Starting on http://localhost:3001

**Reply by dcousens** [reply]
Alternatively, you can define `server.port` in your configuration
```typescript
  config<TypeInfo>({
    server: {
      port: 3001
    },
    // ...
  })

**Comment by tokenosopher** [comment]
What worked for me was setting the following in package.json:

"dev": "set PORT=4000 && keystone dev"
